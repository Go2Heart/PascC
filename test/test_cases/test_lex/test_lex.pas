program quicksort(input,output);  
var  
n,i:integer;  
a:array[0..100000] of integer;  
b:char;  
  
procedure kp(l,r:integer);  
var  
i,j,mid:integer;  
begin  
    b:='a';  
    if l>=r then exit;  
    i:=l;j:=r;mid:=a[(l+r) div 2];  
    repeat  
    begin  
         while a[i]<mid do inc(i);  
         while a[j]>mid do dec(j);  
         if i<=j then  
         begin  
             a[0]:=a[i];a[i]:=a[j];a[j]:=a[0];  
             inc(i);dec(j);  
         end  
    end  
    until i>j;  
    kp(l,j);
    kp(i,r)  
end;  
  
begin  
    readln(n);  
    for i:=1 to n do  
    read(a[i]);  
    kp(1,n);  
    for i:=1 to n do  
    write(a[i]);  
end.  
