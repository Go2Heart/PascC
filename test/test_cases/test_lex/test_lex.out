LexToken(PROGRAM,'program',1,0)
LexToken(ID,'quicksort',1,8)
LexToken(LPAREN,'(',1,17)
LexToken(ID,'input',1,18)
LexToken(COMMA,',',1,23)
LexToken(ID,'output',1,24)
LexToken(RPAREN,')',1,30)
LexToken(SEMI,';',1,31)
LexToken(VAR,'var',2,35)
LexToken(ID,'n',3,41)
LexToken(COMMA,',',3,42)
LexToken(ID,'i',3,43)
LexToken(COLON,':',3,44)
LexToken(INTEGER,'integer',3,45)
LexToken(SEMI,';',3,52)
LexToken(ID,'a',4,56)
LexToken(COLON,':',4,57)
LexToken(ARRAY,'array',4,58)
LexToken(LBRACK,'[',4,63)
LexToken(ICONST,0,4,64)
LexToken(DOTDOT,'..',4,65)
LexToken(ICONST,100000,4,67)
LexToken(RBRACK,']',4,73)
LexToken(OF,'of',4,75)
LexToken(INTEGER,'integer',4,78)
LexToken(SEMI,';',4,85)
LexToken(ID,'b',5,89)
LexToken(COLON,':',5,90)
LexToken(CHAR,'char',5,91)
LexToken(SEMI,';',5,95)
LexToken(PROCEDURE,'procedure',7,102)
LexToken(ID,'kp',7,112)
LexToken(LPAREN,'(',7,114)
LexToken(ID,'l',7,115)
LexToken(COMMA,',',7,116)
LexToken(ID,'r',7,117)
LexToken(COLON,':',7,118)
LexToken(INTEGER,'integer',7,119)
LexToken(RPAREN,')',7,126)
LexToken(SEMI,';',7,127)
LexToken(VAR,'var',8,131)
LexToken(ID,'i',9,137)
LexToken(COMMA,',',9,138)
LexToken(ID,'j',9,139)
LexToken(COMMA,',',9,140)
LexToken(ID,'mid',9,141)
LexToken(COLON,':',9,144)
LexToken(INTEGER,'integer',9,145)
LexToken(SEMI,';',9,152)
LexToken(BEGIN,'begin',10,156)
LexToken(ID,'b',11,168)
LexToken(ASSIGN,':=',11,169)
LexToken(CCONST,"'a'",11,171)
LexToken(SEMI,';',11,174)
LexToken(IF,'if',12,182)
LexToken(ID,'l',12,185)
LexToken(RELOP,'>=',12,186)
LexToken(ID,'r',12,188)
LexToken(THEN,'then',12,190)
LexToken(ID,'exit',12,195)
LexToken(SEMI,';',12,199)
LexToken(ID,'i',13,207)
LexToken(ASSIGN,':=',13,208)
LexToken(ID,'l',13,210)
LexToken(SEMI,';',13,211)
LexToken(ID,'j',13,212)
LexToken(ASSIGN,':=',13,213)
LexToken(ID,'r',13,215)
LexToken(SEMI,';',13,216)
LexToken(ID,'mid',13,217)
LexToken(ASSIGN,':=',13,220)
LexToken(ID,'a',13,222)
LexToken(LBRACK,'[',13,223)
LexToken(LPAREN,'(',13,224)
LexToken(ID,'l',13,225)
LexToken(ADDOP,'+',13,226)
LexToken(ID,'r',13,227)
LexToken(RPAREN,')',13,228)
LexToken(MULDIVANDOP,'div',13,230)
LexToken(ICONST,2,13,234)
LexToken(RBRACK,']',13,235)
LexToken(SEMI,';',13,236)
LexToken(REPEAT,'repeat',14,244)
LexToken(BEGIN,'begin',15,257)
LexToken(WHILE,'while',16,274)
LexToken(ID,'a',16,280)
LexToken(LBRACK,'[',16,281)
LexToken(ID,'i',16,282)
LexToken(RBRACK,']',16,283)
LexToken(RELOP,'<',16,284)
LexToken(ID,'mid',16,285)
LexToken(DO,'do',16,289)
LexToken(ID,'inc',16,292)
LexToken(LPAREN,'(',16,295)
LexToken(ID,'i',16,296)
LexToken(RPAREN,')',16,297)
LexToken(SEMI,';',16,298)
LexToken(WHILE,'while',17,311)
LexToken(ID,'a',17,317)
LexToken(LBRACK,'[',17,318)
LexToken(ID,'j',17,319)
LexToken(RBRACK,']',17,320)
LexToken(RELOP,'>',17,321)
LexToken(ID,'mid',17,322)
LexToken(DO,'do',17,326)
LexToken(ID,'dec',17,329)
LexToken(LPAREN,'(',17,332)
LexToken(ID,'j',17,333)
LexToken(RPAREN,')',17,334)
LexToken(SEMI,';',17,335)
LexToken(IF,'if',18,348)
LexToken(ID,'i',18,351)
LexToken(RELOP,'<=',18,352)
LexToken(ID,'j',18,354)
LexToken(THEN,'then',18,356)
LexToken(BEGIN,'begin',19,372)
LexToken(ID,'a',20,393)
LexToken(LBRACK,'[',20,394)
LexToken(ICONST,0,20,395)
LexToken(RBRACK,']',20,396)
LexToken(ASSIGN,':=',20,397)
LexToken(ID,'a',20,399)
LexToken(LBRACK,'[',20,400)
LexToken(ID,'i',20,401)
LexToken(RBRACK,']',20,402)
LexToken(SEMI,';',20,403)
LexToken(ID,'a',20,404)
LexToken(LBRACK,'[',20,405)
LexToken(ID,'i',20,406)
LexToken(RBRACK,']',20,407)
LexToken(ASSIGN,':=',20,408)
LexToken(ID,'a',20,410)
LexToken(LBRACK,'[',20,411)
LexToken(ID,'j',20,412)
LexToken(RBRACK,']',20,413)
LexToken(SEMI,';',20,414)
LexToken(ID,'a',20,415)
LexToken(LBRACK,'[',20,416)
LexToken(ID,'j',20,417)
LexToken(RBRACK,']',20,418)
LexToken(ASSIGN,':=',20,419)
LexToken(ID,'a',20,421)
LexToken(LBRACK,'[',20,422)
LexToken(ICONST,0,20,423)
LexToken(RBRACK,']',20,424)
LexToken(SEMI,';',20,425)
LexToken(ID,'inc',21,442)
LexToken(LPAREN,'(',21,445)
LexToken(ID,'i',21,446)
LexToken(RPAREN,')',21,447)
LexToken(SEMI,';',21,448)
LexToken(ID,'dec',21,449)
LexToken(LPAREN,'(',21,452)
LexToken(ID,'j',21,453)
LexToken(RPAREN,')',21,454)
LexToken(SEMI,';',21,455)
LexToken(END,'end',22,468)
LexToken(END,'end',23,478)
LexToken(UNTIL,'until',24,488)
LexToken(ID,'i',24,494)
LexToken(RELOP,'>',24,495)
LexToken(ID,'j',24,496)
LexToken(SEMI,';',24,497)
LexToken(ID,'kp',25,505)
LexToken(LPAREN,'(',25,507)
LexToken(ID,'l',25,508)
LexToken(COMMA,',',25,509)
LexToken(ID,'j',25,510)
LexToken(RPAREN,')',25,511)
LexToken(SEMI,';',25,512)
LexToken(ID,'kp',26,518)
LexToken(LPAREN,'(',26,520)
LexToken(ID,'i',26,521)
LexToken(COMMA,',',26,522)
LexToken(ID,'r',26,523)
LexToken(RPAREN,')',26,524)
LexToken(END,'end',27,528)
LexToken(SEMI,';',27,531)
LexToken(BEGIN,'begin',29,538)
LexToken(READLN,'readln',30,550)
LexToken(LPAREN,'(',30,556)
LexToken(ID,'n',30,557)
LexToken(RPAREN,')',30,558)
LexToken(SEMI,';',30,559)
LexToken(FOR,'for',31,567)
LexToken(ID,'i',31,571)
LexToken(ASSIGN,':=',31,572)
LexToken(ICONST,1,31,574)
LexToken(TO,'to',31,576)
LexToken(ID,'n',31,579)
LexToken(DO,'do',31,581)
LexToken(READ,'read',32,590)
LexToken(LPAREN,'(',32,594)
LexToken(ID,'a',32,595)
LexToken(LBRACK,'[',32,596)
LexToken(ID,'i',32,597)
LexToken(RBRACK,']',32,598)
LexToken(RPAREN,')',32,599)
LexToken(SEMI,';',32,600)
LexToken(ID,'kp',33,608)
LexToken(LPAREN,'(',33,610)
LexToken(ICONST,1,33,611)
LexToken(COMMA,',',33,612)
LexToken(ID,'n',33,613)
LexToken(RPAREN,')',33,614)
LexToken(SEMI,';',33,615)
LexToken(FOR,'for',34,623)
LexToken(ID,'i',34,627)
LexToken(ASSIGN,':=',34,628)
LexToken(ICONST,1,34,630)
LexToken(TO,'to',34,632)
LexToken(ID,'n',34,635)
LexToken(DO,'do',34,637)
LexToken(WRITE,'write',35,646)
LexToken(LPAREN,'(',35,651)
LexToken(ID,'a',35,652)
LexToken(LBRACK,'[',35,653)
LexToken(ID,'i',35,654)
LexToken(RBRACK,']',35,655)
LexToken(RPAREN,')',35,656)
LexToken(SEMI,';',35,657)
LexToken(END,'end',36,661)
LexToken(DOT,'.',36,664)
